src/box/lua/tuple.c

```diff
@@ -327,7 +336,12 @@ lbox_tuple_new(lua_State *L)
 		luaT_pushtuple(L, tuple);
 		return 1;
 	}
-
+	if (argc == 1 && lua_type(L, 1) != LUA_TTABLE) {
```

Comment:
AFAIU, there is one more case, when the single argument is passed and we should keep the old behavior:

```lua
tarantool> box.tuple.new(box.tuple.new({1, 2, 3}))
---
- [1, 2, 3]
...
```

I guess this code will make it `[[1, 2, 3]]`.

BTW, regarding the next patch, I would make `box.tuple.new(smth)` and `box.tuple.new(smth, {format = <...>})` return the same shaped tuple for all `smth` inputs (when the compat option is `new`). I didn't verify it, highlighted just in case.

Reply:
Thanks for noticing! Indeed, that's a bug, fixed and added a test case.